binarysort/binarysort.go

```diff
@@ -0,0 +1,126 @@
+// Package binarysort provides types and functions to encode into naturally sorted binary representations.
+// That way, if vA < vB, where vA and vB are two unencoded values of the same type, then eA < eB, where eA and eB
+// are the respective encoded values of vA and vB.
+package binarysort
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/binary"
+	"errors"
+	"math"
+)
+
+// Default Base64 encoder string doesn't preserve lexicographic order. This alternative
+// encoder does.
+const base64encoder = "-0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ_abcdefghijklmnopqrstuvwxyz"
+
+var base64Encoding = base64.NewEncoding(base64encoder).WithPadding(base64.NoPadding)
+
+// AppendBool takes a bool and returns its binary representation.
+func AppendBool(buf []byte, x bool) []byte {
+	if x {
+		return append(buf, 1)
+	}
+	return append(buf, 0)
+}
+
+// DecodeBool takes a byte slice and decodes it into a boolean.
+func DecodeBool(buf []byte) bool {
+	return buf[0] == 1
```

Comment:
We need a length check here like in `DecodeUint64`. Or document that `DecodeBool` expects non-empty `buf`.